src/main/java/org/stellar/sdk/LiquidityPoolConstantProductParameters.java

```diff
@@ -0,0 +1,96 @@
+package org.stellar.sdk;
+
+import com.google.common.base.Objects;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Base LiquidityPoolConstantProductParameters class.
+ * @see <a href="https://www.stellar.org/developers/learn/concepts/assets.html" target="_blank">Assets</a>
+ */
+public final class LiquidityPoolConstantProductParameters extends LiquidityPoolParameters {
+  protected final Asset assetA;
+  protected final Asset assetB;
+  protected final int fee;
+
+  /**
+   * LiquidityPoolConstantProductParameters represents the parameters specifying a constant-product liquidity pool.
+   *
+   * @param a First asset in the liquidity pool
+   * @param b Second asset in the liquidity pool
+   * @param feeBP Fee amount in base-points
+   */
+  public LiquidityPoolConstantProductParameters(Asset a, Asset b, int feeBP) {
+    checkNotNull(a, "asset A cannot be null");
+    checkNotNull(b, "asset B cannot be null");
+    checkNotNull(feeBP, "fee cannot be null");
+    assetA = a;
+    assetB = b;
+    fee = feeBP;
+  }
+
+  public Asset getAssetA() {
+    return assetA;
+  }
+
+  public Asset getAssetB() {
+    return assetB;
+  }
+
+  public int getFee() {
+    return fee;
+  }
+
+  @Override
+  public final boolean equals(Object object) {
+    if (object == null || !this.getClass().equals(object.getClass())) {
+        return false;
+    }
+
+    LiquidityPoolConstantProductParameters o = (LiquidityPoolConstantProductParameters) object;
+    return Objects.equal(this.getAssetA(), o.getAssetA())
+      && Objects.equal(this.getAssetB(), o.getAssetB())
+      && Objects.equal(this.getFee(), o.getFee());
+  }
+
+  /**
+   * Generates XDR object from a given LiquidityPoolParameters object
+   */
+  @Override
+  public final org.stellar.sdk.xdr.LiquidityPoolParameters toXdr() {
```

Comment:
similar to the comment below you can omit the "org.stellar.sdk.xdr" prefix everywhere in this file

Reply:
I don't think we can here, because there's a collision with `LiquidityPoolParameters` in the current package.

Reply:
yeah, you're right. sorry you can ignore this comment